src/uhitm.c

```diff
@@ -1501,6 +1501,13 @@ hmon_hitmon_do_hit(
     struct monst *mon,
     struct obj *obj)
 {
+    if (!(is_silent(gy.youmonst.data) && helpless(mon))
+        && hmd->hand_to_hand && rn2(Stealth ? 10 : 5)) {
+        int base_combat_noise = combat_noise(&mons[gu.urace.mnum]);
+        if (Stealth)
```

Comment:
With the refactor, don't all 3 cases (um, mm, mu) go through this code path? So the hero's stealth should not affect monster-vs-monster or monster-vs-player cases.

Reply:
> With the refactor, don't all 3 cases (um, mm, mu) go through this code path?

I didn't think so, I thought it just broke up `hmon_hitmon` into chunks.  `hitmu` and `hitmm` are still separate functions (with their own similar wakeup blocks).

Reply:
There are some unconditional `You()` calls scattered around the various `hmon_hitmon_foo` functions so yeah, I don't think all the cases go through it.  Maybe it's the first step towards that, but I don't think it's there yet.